### ISSUE TITLE
bug: Error when interacting with elements on iOS

### ISSUE THREAD

**Issue by viniborg** [appium/appium#19962]
### Do I have the most recent component updates?

- [X] I use the most recent available driver/plugin and server versions

### Is the component officially supported by the Appium team?

- [X] I have verified the component repository is present under the Appium organization in GitHub

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

[appiumLog.txt](https://github.com/appium/appium/files/14842221/appiumLog.txt)

I have an automation for a mobile device using iOS. The automation performs several interactions successfully until logging into the application and viewing its home screen. After that, a "click" command is executed, which apparently causes the server to "freeze," as per the attached log. Appium generates the following error until the session expires. However, the simulator remains active, and there were no connection problems or anything similar.

`Encountered internal error running command: NoSuchDriverError: A session is either terminated or not started [AppiumDriver@88f0] at asyncHandler (/usr/local/lib/node_modules/appium/node_modules/@appium/base-driver/lib/protocol/protocol.js:315:15) [AppiumDriver@88f0] at /usr/local/lib/node_modules/appium/node_modules/@appium/base-driver/lib/protocol/protocol.js:518:15) [AppiumDriver@88f0] at Layer.handle [as handle_request] (/usr/local/lib/node_modules/appium/node_modules/express/lib/router/layer.js:95:5) [AppiumDriver@88f0] at next (/usr/local/lib/node_modules/appium/node_modules/express/lib/router/route.js:144:13) [AppiumDriver@88f0] at Route.dispatch (/usr/local/lib/node_modules/appium/node_modules/express/lib/router/route.js:114:3) [AppiumDriver@88f0] at Layer.handle [as handle_request] (/usr/local/lib/node_modules/appium/node_modules/express/lib/router/layer.js:95:5) [AppiumDriver@88f0] at /usr/local/lib/node_modules/appium/node_modules/express/lib/router/index.js:284:15) [AppiumDriver@88f0] at param (/usr/local/lib/node_modules/appium/node_modules/express/lib/router/index.js:365:14) [AppiumDriver@88f0] at param (/usr/local/lib/node_modules/appium/node_modules/express/lib/router/index.js:376:14) [AppiumDriver@88f0] at Function.process_params (/usr/local/lib/node_modules/appium/node_modules/express/lib/router/index.js:421:3) [AppiumDriver@88f0] at next (/usr/local/lib/node_modules/appium/node_modules/express/lib/router/index.js:280:10) [AppiumDriver@88f0] at logger (/usr/local/lib/node_modules/appium/node_modules/morgan/index.js:144:5) [AppiumDriver@88f0] at Layer.handle [as handle_request] (/usr/local/lib/node_modules/appium/node_modules/express/lib/router/layer.js:95:5) [AppiumDriver@88f0] at trim_prefix (/usr/local/lib/node_modules/appium/node_modules/express/lib/router/index.js:328:13) [AppiumDriver@88f0] at /usr/local/lib/node_modules/appium/node_modules/express/lib/router/index.js:286:9) [AppiumDriver@88f0] at Function.process_params (/usr/local/lib/node_modules/appium/node_modules/express/lib/router/index.js:346:12) [AppiumDriver@88f0] at next (/usr/local/lib/node_modules/appium/node_modules/express/lib/router/index.js:280:10) [AppiumDriver@88f0] at jsonParser (/usr/local/lib/node_modules/appium/node_modules/body-parser/lib/types/json.js:113:7) [AppiumDriver@88f0] at Layer.handle [as handle_request] (/usr/local/lib/node_modules/appium/node_modules/express/lib/router/layer.js:95:5) [AppiumDriver@88f0] at trim_prefix (/usr/local/lib/node_modules/appium/node_modules/express/lib/router/index.js:328:13) [AppiumDriver@88f0] at /usr/local/lib/node_modules/appium/node_modules/express/lib/router/index.js:286:9) [AppiumDriver@88f0] at Function.process_params (/usr/local/lib/node_modules/appium/node_modules/express/lib/router/index.js:346:12)`

The problem seems to be reproduced on iOS versions 17.0.
I have tested with several other combinations of versions, such as:

Appium server: 2.4.1
xcuitest@5.16.0
WebDriverAgent: 5.15.8

Appium server: 2.5.1
xcuitest@7.9.1
WebDriverAgent: 8.4.0

I'm using java-client 9.2.1.
```
 <groupId>io.appium</groupId>
            <artifactId>java-client</artifactId>
            <version>9.2.1</version>
```

The piece of code where the issue seems to occur is:

```
@iOSXCUITFindBy(iOSNsPredicate = "label == \"Serviços\" AND type == \"XCUIElementTypeButton\"")
@AndroidFindBy(id = "constraintlayout_container_service")
private WebElement services;


public void accessServices() {
        waitForVisibilityOf(services).click();
    }

```

### Expected Behavior

To be able to interact with the elements.

### Minimal Reproducible Example

```
@iOSXCUITFindBy(iOSNsPredicate = "label == \"Serviços\" AND type == \"XCUIElementTypeButton\"")
@AndroidFindBy(id = "constraintlayout_container_service")
private WebElement services;


public void accessServices() {
        waitForVisibilityOf(services).click();
    }

```

### Environment

- Operating system: Mac Book Apple M1, with Ventura 13.6.4
- Appium server version (output of `appium --version`): 2.5.1
- Appium driver(s) and their version(s): xcuitest@7.9.1 [installed (npm)]
- Appium plugin(s) and their version(s):
- Node.js version (output of `node --version`): v16.17.0
- `npm` version (output of `npm --version`): 8.15.0
- Last component(s) version which did _not_ exhibit the problem:
- Platform and version under test:
- Real device or emulator/simulator: iPhone 15 Pro Max Simulator (17.0)

**Comment by jlipps** [comment]
Nothing in the log seems out of place, except for the recurring request to /timeouts for a nonexistent session (this would be something that's happening on your computer, like maybe an old Inspector window open). One call to findelement does take a long time and your client disconnects in the middle of it, so maybe you should increase your client http timeout?

**Comment by viniborg** [comment]
@jlipps Do you have any idea why there is a long delay for the click to happen? Apparently, based on the logs, the element is being found. However, executing the click takes minutes.

![Captura de Tela 2024-04-02 às 16 47 23](https://github.com/appium/appium/assets/83841507/98cd5301-dc46-4181-84ed-64f5555ba2e3)

**Comment by viniborg** [comment]
I set a timeout of 5 minutes, and still, the session is terminated without executing the command.

**Comment by mykola-mokhnach** [comment]
Try to also get the device log for the particular time period. I was observing such freezes when XCTest was unable to interact with the application under test because it was not idling.

**Comment by viniborg** [comment]
@mykola-mokhnach 
Here is the new log from the server and the device, both corresponding to the same session.
Adding to the information, I have already performed tests on emulators and real devices. The log above is from a simulator.

[deviceLog03.txt](https://github.com/appium/appium/files/14852900/deviceLog03.txt)
[appiumLog03.txt](https://github.com/appium/appium/files/14852902/appiumLog03.txt)

**Comment by mykola-mokhnach** [comment]
unfortunately the device log does not contain any useful input. I would try to debug the application under test for the above scenario to see what exactly it is doing after this click happens, so it blocks the main event loop. Otherwise there is nothing you could do from the driver side.

**Comment by mykola-mokhnach** [comment]
Closed as third party issue

**Comment by viniborg** [comment]
After analyzing the problem encountered, we identified the root cause. It seems that the delay in the application was due to the slow transition into idle mode. Initially, we were using the parameter `driver.setSetting(Setting.WAIT_FOR_IDLE_TIMEOUT, idleTimeout)` with a value of 2000 milliseconds. However, this time was not sufficient for the application to fully enter idle mode. Consequently, between the identification of the element and the interaction with it, the structure of the elements had changed, resulting in the issue at hand.
